aioaws/core.py

```diff
@@ -23,12 +23,108 @@
 _AUTH_ALGORITHM = 'AWS4-HMAC-SHA256'
 
 
+def _aws4_date_stamp(dt: datetime) -> str:
+    return dt.strftime('%Y%m%d')
+
+
+def _aws4_x_amz_date(dt: datetime) -> str:
+    return dt.strftime('%Y%m%dT%H%M%SZ')
+
+
+def _aws4_reduce_signature(key: bytes, msg: str) -> bytes:
+    return hmac.new(key, msg.encode(), hashlib.sha256).digest()
+
+
+class _AWSv4Auth:
```

Comment:
```suggestion
class AWSv4Auth:
```

We don't need to add it to `__all__`, but I think we can remove the `_` shouting "THIS IS VERY PRIVATE".

Reply:
addressed